rs/http_endpoints/public/src/read_state/subnet.rs

```diff
@@ -13,28 +16,30 @@ use axum::{
 use crossbeam::atomic::AtomicCell;
 use http::Request;
 use hyper::StatusCode;
-use ic_crypto_tree_hash::{Label, Path, TooLongPathError, sparse_labeled_tree_from_paths};
+use ic_crypto_tree_hash::Path;
 use ic_interfaces_state_manager::StateReader;
 use ic_nns_delegation_manager::{CanisterRangesFilter, NNSDelegationReader};
 use ic_replicated_state::ReplicatedState;
 use ic_types::{
-    CanisterId, PrincipalId,
-    messages::{
-        Blob, Certificate, HttpReadStateContent, HttpReadStateResponse, HttpRequest,
-        HttpRequestEnvelope, ReadState,
-    },
+    CanisterId, PrincipalId, SubnetId,
+    messages::{HttpReadStateContent, HttpRequest, HttpRequestEnvelope, ReadState},
 };
 use std::{
     convert::{Infallible, TryFrom},
     sync::Arc,
 };
 use tower::util::BoxCloneService;
 
-#[derive(Copy, Clone, Debug)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
 pub enum Version {
-    // Endpoint with the NNS delegation using the flat format of the canister ranges.
+    /// Endpoint with the NNS delegation using the flat format of the canister ranges.
+    /// `/subnet/<subnet_id>/canister_ranges` path is allowed
     V2,
-    // Endpoint with the NNS delegation will all canister ranges pruned out.
+    /// Endpoint with the NNS delegation using the tree format of the canister ranges.
```

Comment:
Didn't we conclude that we don't expose any canister ranges for subnet endpoints?